From d509574f06909666742908c8375d72d598d7df4d Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Tue, 9 Jul 2024 17:25:37 +0530
Subject: [PATCH] mtd: block2mtd: Handle block2mtd parameters

Add fix to handle block2mtd parameters for NAND flash with secure storage.
block2mtd parsing totally 5(0 to 4) parameters but array size is only 4(token[4]),
so when try to access last parameter kernel is crashing, added fix to accommodate
the same.

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: Ideac0766612b0e89d2afe16a64b3bdab595a639f
---
 drivers/mtd/devices/block2mtd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/devices/block2mtd.c b/drivers/mtd/devices/block2mtd.c
index 0c2d94655648..3c6146dec684 100644
--- a/drivers/mtd/devices/block2mtd.c
+++ b/drivers/mtd/devices/block2mtd.c
@@ -34,7 +34,7 @@
 char parent_mtd_name[80];
 
 /* Maximum number of comma-separated items in the 'block2mtd=' parameter */
-#define BLOCK2MTD_PARAM_MAX_COUNT 3
+#define BLOCK2MTD_PARAM_MAX_COUNT 5
 
 /* Info for the block device */
 struct block2mtd_dev {
@@ -438,7 +438,7 @@ static int block2mtd_setup2(const char *val)
 	 * 80 for parent mtd device name, 8 for timeout */
 	char buf[80 + 12 + 80 + 80 + 8];
 	char *str = buf;
-	char *token[BLOCK2MTD_PARAM_MAX_COUNT + 1];
+	char *token[BLOCK2MTD_PARAM_MAX_COUNT];
 	char *name;
 	char *label = NULL;
 	size_t erase_size = PAGE_SIZE;
@@ -453,7 +453,7 @@ static int block2mtd_setup2(const char *val)
 	strcpy(str, val);
 	kill_final_newline(str);
 
-	for (i = 0; i < BLOCK2MTD_PARAM_MAX_COUNT + 1; i++)
+	for (i = 0; i < BLOCK2MTD_PARAM_MAX_COUNT; i++)
 		token[i] = strsep(&str, ",");
 
 	if (str) {
-- 
2.34.1

